src/basic/meson.build

```diff
@@ -274,6 +275,7 @@ libbasic = static_library(
         fundamental_sources,
         include_directories : basic_includes,
         dependencies : [libcap,
+                        libcurl,
```

Comment:
so this shouldn't be pulled into the required set of libs for us like this.

first of all, this should be in shared/ not basic/. And it should really we be a dlopen() style dependency, see how this is done for libraries such as libfido2 or libqrencode and similar.

i.e. i think we should make use of libcurl if we have it, but if we don't that's entirely fine, we can just work anyway (though without doh).

Reply:
Moved to shared/, I did something to include libcurl in libshared_deps conditionally but I'm unsure if that is correct or not, could you please advise on it?